lib/index.js

```diff
@@ -47,8 +49,190 @@ api.validateConfiguration = (validatorConfig, callback) =>
   validate('ledger-validator-multiproof', validatorConfig, callback);
 
 api.validateEvent = (event, validatorConfig, callback) => {
-  // determine which proof types should be checked
-
-  // FIXME: Implement
-  callback();
+  async.auto({
+    schema: callback => callback(),
+    signatures: ['schema', (results, callback) =>
+      _validateSignatures(event, callback)]
+  }, err => callback(err));
 };
+
+function _validateSignatures(event, callback) {
+
+  // FIXME: could these parameters just be in the ledger configuration instead
+  // of inventing `ParameterAlgorithm`?
+  const equihashParameterAlgorithm = {
+    VeresOne2017: {n: 64, k: 3}
+  };
+  const signatureType = ['LinkedDataSignature2015', 'EquihashProof2017'];
+
+  // FIXME: authorizationCapability[0].permission is `UpdateDidDocument`, how
+  // does this validator know that the current operation is UpdateDidDocument?
```

Comment:
The event operation would include that. It may be that "update" would be used for the operation and not "UpdateDidDocument" and we'd map these ... we should chat a bit more with @msporny on this one.

Reply:
wait for input from @msporny on this.

Reply:
solution: make sure that webledger operation is 'create' and the permission is 'UpdateDidDocument'

Reply:
Done.